docs/source/conf.py

```diff
@@ -166,7 +166,18 @@
     ),
 ]
 
-intersphinx_mapping = {"python": ("https://docs.python.org/3", None)}
+intersphinx_mapping = {
+    "python": ("https://docs.python.org/3", None),
+    "distributed": ("https://distributed.dask.org/en/stable", None),
+    "lightgbm": ("https://lightgbm.readthedocs.io/en/latest", None),
+    "matplotlib": ("https://matplotlib.org/stable", None),
+    "numpy": ("https://numpy.org/doc/stable", None),
+    "scipy": ("https://docs.scipy.org/doc/scipy", None),
+    "sklearn": ("https://scikit-learn.org/stable", None),
+    "torch": ("https://pytorch.org/docs/stable", None),
+    "pandas": ("https://pandas.pydata.org/docs", None),
```

Comment:
[Notes] The URL seems to be correct, but the `pd.Dataframe` is not linked to the pandas docs. Maybe we need to update the type hint in a follow-up PR.
<img width="1095" alt="image" src="https://user-images.githubusercontent.com/3255979/216489542-55154caa-6064-4dec-80b0-568b90eb86a1.png">